util/src/entities/ConnectedAccount.ts

```diff
@@ -1,11 +1,14 @@
-import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
-import { BaseClass } from "./BaseClass";
+import { Column, Entity, JoinColumn, ManyToOne, PrimaryColumn, RelationId } from "typeorm";
+import { BaseClassWithoutId } from "./BaseClass";
 import { User } from "./User";
 
 export interface PublicConnectedAccount extends Pick<ConnectedAccount, "name" | "type" | "verified"> {}
 
 @Entity("connected_accounts")
-export class ConnectedAccount extends BaseClass {
+export class ConnectedAccount extends BaseClassWithoutId {
+	@PrimaryColumn()
+	id: string;
```

Comment:
Why do you manually create the ID? If Mongo support is ever fixed, we'll have to repeat the mongo object ID annotation here iirc ( unless the typeorm api has changed for mongo? )

Reply:
> Why do you manually create the ID? If Mongo support is ever fixed, we'll have to repeat the mongo object ID annotation here iirc ( unless the typeorm api has changed for mongo? )

this is the id for the user on the connection site, like github user id, youtube channel id, etc.

Reply:
Ahhhhh I understand. I think it might be better to include our snowflake IDs purely so we have creation time for the connection, since we have that for every other entity. Might be a bit annoying renaming the external ID from `external_id` or whatever to `id` before you send to client though

Reply:
> Ahhhhh I understand. I think it might be better to include our snowflake IDs purely so we have creation time for the connection, since we have that for every other entity. Might be a bit annoying renaming the external ID from `external_id` or whatever to `id` before you send to client though

That's what I wanted to do, and was asking Arcane about in Discord, but I couldn't figure out the best way to do it. although `external_id` is a good idea for what we could name the column. 

Ref to Discord chat: https://discord.com/channels/806142446094385153/938936139732090931/1005934691930607746)

Reply:
Just realised it's not guaranteed that IDs from external providers will be unique anyway, so you can't use them as the primary key here anyway. Just extend BaseClass normally and have an `external_id` key, and when you send the connections to the client in `users/@me/connections` do smth like `connections.map(x => ({ ...x, id: x.external_id, external_id: undefined }))`